manifests/init.pp

```diff
@@ -312,6 +314,14 @@
       fail('You must provide a value for multicast_address')
   }
 
+  # $rrp_mode value of 'none' isn't allowed with multiple rings
+  if ( ( is_array($multicast_address) and size($multicast_address) > 1 )
+  or ( is_array($unicast_addresses) and is_array($unicast_addresses[0]) and size($unicast_addresses[0]) > 1 )
+  or ( is_array($quorum_members) and is_array($quorum_members[0]) and size($quorum_members[0]) > 1 ) )
+  and $rrp_mode == 'none' {
+    fail('You must set rrp_mode to active or passive with multiple rings')
```

Comment:
Would using validate_cmd from puppetlabs-stdlib be okay?
validate_cmd(template($corosync_conf), 'corosync -t', 'Corosync failed to validate  /etc/corosync/corosync.conf')